pkg/kubelet/kubelet.go

```diff
@@ -1980,6 +1984,9 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 				klog.V(4).InfoS("SyncLoop (PLEG): pod does not exist, ignore irrelevant event", "event", e)
 			}
 		}
+		if e.Type == pleg.ContainerRemoved {
+			kl.deletePodSandbox(e.ID)
```

Comment:
This no longer belongs here, this belongs in pod worker `syncTerminatedPod`.  As a result of the changes in #102344 I'm pretty sure it is already there, so this is an unnecessary change and should be removed.

Reply:
I looked into the PR and it doesn't seem to fix this scenario atleast from the looks of it. I can test it once to confirm. The reason why i don't feel the PR fixes is because in the syncTerminatedPod we call KillPod which only stops the sandboxes but doesn't remove them and leaves it to the GC to do that.